backend/src/api/mod.rs

```diff
@@ -1,7 +1,10 @@
-pub mod account;
-pub mod budget;
-pub mod currency;
+use actix_web::HttpResponse;
+
+use crate::api::error::api::ApiError;
+
 pub mod error;
+pub mod pagination;
+pub mod routes;
 pub mod status;
-pub mod transaction;
-pub mod user;
+
+pub type ApiResponse = Result<HttpResponse, ApiError>;
```

Comment:
Maybe replace with:
```suggestion
pub type ApiResponse = Result<dyn Response, ApiError>;
```
Because using `impl Response` isn't stable for type aliases.